**Erdem Muslu / 2111051015 - Intro_ML_Assignment_2**

In [15]:
import kagglehub
kagglehub.login()

dansbecker_home_data_for_ml_course_path = kagglehub.dataset_download('dansbecker/home-data-for-ml-course')
print('Data source import complete.')

Data source import complete.
Kaggle credentials set.
Kaggle credentials successfully validated.


In [16]:
import pandas as pd
import numpy as np

iowa_file_path = dansbecker_home_data_for_ml_course_path + '/train.csv'
housing_data = pd.read_csv(iowa_file_path)
print("Datasets loaded successfully!")

Datasets loaded successfully!


# Question 1: Initial Data Inspection

In [17]:
# Display the first 7 rows of the DataFrame
print(housing_data.head(7))

   Id  MSSubClass MSZoning  LotFrontage  LotArea Street Alley LotShape  \
0   1          60       RL         65.0     8450   Pave   NaN      Reg   
1   2          20       RL         80.0     9600   Pave   NaN      Reg   
2   3          60       RL         68.0    11250   Pave   NaN      IR1   
3   4          70       RL         60.0     9550   Pave   NaN      IR1   
4   5          60       RL         84.0    14260   Pave   NaN      IR1   
5   6          50       RL         85.0    14115   Pave   NaN      IR1   
6   7          20       RL         75.0    10084   Pave   NaN      Reg   

  LandContour Utilities  ... PoolArea PoolQC  Fence MiscFeature MiscVal  \
0         Lvl    AllPub  ...        0    NaN    NaN         NaN       0   
1         Lvl    AllPub  ...        0    NaN    NaN         NaN       0   
2         Lvl    AllPub  ...        0    NaN    NaN         NaN       0   
3         Lvl    AllPub  ...        0    NaN    NaN         NaN       0   
4         Lvl    AllPub  ...    

Looking at the first few rows, we can identify several columns. For example:


*   Numerical feature related to size or area: LotArea (represents the area of the lot).

*   Numerical feature related to price: SalePrice (shows the price at which the house was sold).

*   Categorical feature: HouseStyle (indicates the style of the house, such as 1Story, 2Story, etc.).






# Question 2: Dataset Dimensions and Data Types

In [19]:
# Print the dimensions of the dataset
print("Dataset dimensions:", housing_data.shape)

Dataset dimensions: (1460, 81)


The output is (1460, 81), it means there are 1,460 houses (data entries) and 81 features (characteristics) recorded for each house. The property used here is .shape.

# Question 3: Descriptive Statistics of 'LotArea'

In [22]:
# Calculate descriptive statistics for the 'LotArea' column
mean_lot = housing_data['LotArea'].mean()
median_lot = housing_data['LotArea'].median()
std_lot = housing_data['LotArea'].std()
min_lot = housing_data['LotArea'].min()
max_lot = housing_data['LotArea'].max()

print("Mean LotArea:", mean_lot)
print("Median LotArea:", median_lot)
print("Standard Deviation of LotArea:", std_lot)
print("Minimum LotArea:", min_lot)
print("Maximum LotArea:", max_lot)

Mean LotArea: 10516.828082191782
Median LotArea: 9478.5
Standard Deviation of LotArea: 9981.264932379147
Minimum LotArea: 1300
Maximum LotArea: 215245


When we look at the LotArea data, the mean value (10,516.83) is higher than the median (9,478.5). This indicates that the data is right-skewed. In other words, larger lots are pulling the mean upward. Additionally, the high standard deviation (9,981.26) and the wide range of lot sizes, varying from 1,300 to 215,245, suggest that the distribution is not symmetrical.

# Question 4: Value Counts of 'HouseStyle'

In [24]:
# Get the value counts for the 'HouseStyle' column
print(housing_data['HouseStyle'].value_counts())

HouseStyle
1Story    726
2Story    445
1.5Fin    154
SLvl       65
SFoyer     37
1.5Unf     14
2.5Unf     11
2.5Fin      8
Name: count, dtype: int64


*1Story* appears most frequently, then that is the most common house style. . The output of .value_counts() shows how many times each house style appears in the dataset. This method gives us a clear picture of the distribution of house styles across the dataset.

# Question 5: Interpreting 'SalePrice' Statistics

In [27]:
# Calculate descriptive statistics for the 'SalePrice' column
print(housing_data['SalePrice'].describe())

count      1460.000000
mean     180921.195890
std       79442.502883
min       34900.000000
25%      129975.000000
50%      163000.000000
75%      214000.000000
max      755000.000000
Name: SalePrice, dtype: float64




*   Average (Mean) SalePrice: $180,921.20

*   Range: $755,000 - $34,900 = $720,100


The standard deviation of 79,442.50 indicates that house prices are spread out over a wide range around the mean (180,921.20). The fact that the standard deviation is large relative to the mean suggests significant variability in house prices, meaning the dataset includes both cheaper and more expensive homes. This reflects the diversity in the housing market.